### PR TITLE
Hardware LAG platform setting for l3ls_evpn role

### PR DESCRIPTION
**Pull request by onurgashi**
## Change Summary

I have crafted a draft for this, the idea is to have this command under platform settings and use it for R series.
- Keeping in mind I put default true, this is open for discussion as well.
- I'd be happy if I can get some feedback on the code
- I also tried to use arista.avd.default(), but need some help with it.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes)
- [x] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

#530 

## Component(s) name
platform.j2

All device types will include this.
default.yml for l3ls_evpn

```
{# platform #}
{% set ns = namespace() %}
{% for platform_setting in platform_settings %}
{%     set ns.platform_defined = false %}
{%     if switch.platform in platform_setting.platforms %}
{%         set ns.platform_defined = true %}
{%         if platform_setting.lag_hardware_only is arista.avd.defined %}
  sand:
    lag:
      hardware_only: {{ platform_setting.lag_hardware_only }}
{%         endif %}
{%         break %}
{%     endif %}
{% endfor %}
{% for platform_setting in platform_settings if ns.platform_defined == false %}
{%     if 'default' in platform_setting.platforms %}
{%         if platform_setting.lag_hardware_only is arista.avd.defined %}
  sand:
    lag:
      hardware_only: {{ platform_setting.lag_hardware_only }}
{%         endif %}
{%         break %}
{%     endif %}
{% endfor %}
```

## Proposed changes

```
platform_settings:
  - platforms: [ default ]
    reload_delay:
      mlag: 300
      non_mlag: 330
  - platforms: [ 7800R3, 7500R3, 7500R, 7280R3, 7280R2, 7280R ]
    tcam_profile: vxlan-routing
    **lag_hardware_only: true**
    reload_delay:
      mlag: 780
      non_mlag: 1020
```

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/base/platform.j2 [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+{% for platform_setting in platform_settings if ns.platform_defined == false %}
+{%     if 'default' in platform_setting.platforms %}
+{%         if platform_setting.lag_hardware_only is arista.avd.defined %}
+  sand:
+    lag:
+      hardware_only: {{ platform_setting.lag_hardware_only }}
+{%         endif %}
+{%         break %}
+{%     endif %}
+{% endfor %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/base/platform.j2]: missing newline

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/defaults/main.yml [hunk 2]
```diff
@@ -88,6 +88,7 @@ platform_settings:
       non_mlag: 330
   - platforms: [ 7800R3, 7500R3, 7500R, 7280R3, 7280R2, 7280R ]
     tcam_profile: vxlan-routing
+    lag_hardware_only: true
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/defaults/main.yml:91]: Changing this default should be considered a breaking change and must be flagged in release-notes. @carlbuchmann

**Review by ClausHolbechArista — APPROVED** [review]
Please fix missing newline, but apart from that it is good.

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/base/platform.j2 (hunk 1 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/base/platform.j2]: Fixed the newline, thanks Claus.

**Comment by ClausHolbechArista** [comment]
LGTM

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/defaults/main.yml (hunk 2 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/defaults/main.yml:91]: noted

**Pull request merged**